boardswarm/src/qcomlt_debug_board.rs

```diff
@@ -0,0 +1,321 @@
+                    seqnum,
+                    properties,
+                ));
+
+                return true;
+            }
+        }
+        false
+    }
+
+    fn remove(&mut self, device: &Device) {
+        self.registrations.remove(device.syspath());
+    }
+}
+
+#[instrument(skip(r, properties))]
+async fn setup_subdevices(
+    r: ActuatorRegistrations,
+    syspath: PathBuf,
+    node: PathBuf,
+    seqnum: u64,
+    mut properties: Properties,
+) {
+    info!(
+        "Setting up QCOMLTDebugBoard actuators for {}",
+        node.display()
+    );
+
+    // Open the serial port
+    let mut port = match tokio_serial::new(node.to_string_lossy(), 115200).open_native_async() {
+        Ok(port) => port,
+        Err(e) => {
+            warn!("Failed to open serial port: {e}");
+            return;
+        }
+    };
+
+    // TODO: Add actuator name as a property to each actuator
+
+    // Now we are connected, actually register the actuators
+    // Power actuator
+    // TODO: The registered name is the serial port name rather than the node ?!?!?!
+    let power_actuator = QCOMLTDebugBoardActuator::new(port, "power".to_string());
+    r.register(&syspath, seqnum, power_actuator, properties);
+
+    /*
+    TODO:
+
+    2025-02-06T14:29:25.710019Z  INFO start: boardswarm::qcomlt_debug_board: Reserving console /dev/ttyACM0 for QCOMLT Debug Board 75eaaac075a445150202022343a460ff
+    2025-02-06T14:29:25.710237Z  INFO setup_subdevices{node="/dev/ttyACM0" seqnum=3}: boardswarm::qcomlt_debug_board: Setting up QCOMLTDebugBoard actuators for /dev/ttyACM0
+
+    2025-02-06T14:29:25.713803Z  INFO setup_subdevices{node="/dev/ttyACM0" seqnum=3}: boardswarm::qcomlt_debug_board: Ignoring outdated registration (seqnum: 3)
+    2025-02-06T14:29:25.713881Z  INFO setup_subdevices{node="/dev/ttyACM2" seqnum=1}: boardswarm::qcomlt_debug_board: Ignoring outdated registration (seqnum: 1)
+
+        */
+
+    // TODO: USB actuator
+    // TODO: Power button actuator
+    // TODO: Reset button actuator
+
+    // TODO: power measurement (voltage, current). The syntax is:
+    /*
+    0mV 0mA
+    [Pp]ower [Uu]sb [s]tatus pwr[Bb]tn [Rr]stbtn
+    */
+}
+
+// TODO add other debug board commands
+enum QCOMLTDebugBoardCommand {
+    PowerOn(tokio::sync::oneshot::Sender<Result<(), std::io::Error>>),
+    PowerOff(tokio::sync::oneshot::Sender<Result<(), std::io::Error>>),
+}
+
+async fn process(
+    mut port: tokio_serial::SerialStream,
+    mut exec: tokio::sync::mpsc::Receiver<QCOMLTDebugBoardCommand>,
+) {
+    while let Some(command) = exec.recv().await {
+        match command {
+            /* TODO: Create a standalone rust library for the QCOMLTDebugBoard
+            similar to https://github.com/boardswarm/mediatek-brom rather than
+            sending raw bytes here. */
+            QCOMLTDebugBoardCommand::PowerOn(sender) => {
+                info!("Sending PowerOn command");
+                let r = port.write_all(b"P\n").await;
```

Comment:
Not sure if newline needed. Need to check.

Reply:
judging by the firmware code you don't.

Reply:
fixed that.